src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -100,34 +97,21 @@ protected Process createProcess() throws IOException {
     }
 
     private Process createRunasProcess(String... commands) throws IOException {
-        String username = userDecorator.getUser();
-
-        byte[] credBlob = WindowsCredUtils.read(username);
-        ByteBuffer bb = ByteBuffer.wrap(credBlob);
-        CharBuffer cb = WindowsCredUtils.getCharsetForSystem().decode(bb);
-
-        // Prepend runas arguments to use it for running commands as different user
-        List<String> args = new ArrayList<>();
-        args.add(runasExePath);
-        args.add("-u:" + username);  // runas username
-        args.add("-p:" + cb);  // plain text password. TODO revisit this because it exposes plaintext password
-        args.add("-l:off");  // disable logging
-        args.add("-b:0"); // set exit code base number to 0
-        args.add("-w:" + dir); // set workdir explicitly
-        // runas is un-escaping customer-provided quotes and escape characters, so we need to escape them
-        // first so they unwrap correctly.
-        List<String> cmd = Arrays.stream(commands).map((s) ->
-                s.replace("\\\"", "\\\\\"").replace("\"", "\\\""))
-                .collect(Collectors.toList());
-        args.addAll(cmd);
-
-        Arrays.fill(cb.array(), (char) 0);  // zero-out temporary buffers
-        Arrays.fill(bb.array(), (byte) 0);
-
-        ProcessBuilder pb = new ProcessBuilder();
-        pb.environment().putAll(environment);
-        Process p = pb.directory(dir).command(args).start();
-        args.clear();  // best effort to clear password presence
+        // Expect username in format: DOMAIN\UserName
+        String[] usernameParts = userDecorator.getUser().split("\\\\");  // split by single backslash
```

Comment:
Single backslash is `\\` right?

Reply:
Right. And backslash itself needs escaping...

Reply:
What are you escaping for?

Reply:
If the user is `DOMAIN\user` then you would need to split on `\` not `\\`

Reply:
Yes the regex for single backslash is `\\`, as a Java string literal, that is `"\\\\"`.